### ISSUE TITLE
Are there In-File Configuration Directives for dialect?

### ISSUE THREAD

**Issue by galenseilisnh** [sqlfluff/sqlfluff#6657]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

I noticed that this **doesn't** work:

```sql
-- sqlfluff:dialect:tsql
```

### Use case

This would just allow using SQLFluff on a directory of queries for which each query may be from a possibly-distinct dialect.

### Dialect

Any

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by keraion** [comment]
The noted inline configuration should be working at least as of 3.3.1:

Here without the configuration (a typo to not activate the dialect change), referencing a default ansi dialect
```sh
> sqlfluff parse issues/66/sqlfluff_6657/test.sql
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    inline_comment:                                           '-- duckdb specific syntax'
[L:  1, P: 26]      |    newline:                                                  '\n'
[L:  2, P:  1]      |    inline_comment:                                           '-- xsqlfluff:core:dialect:duckdb'
[L:  2, P: 33]      |    newline:                                                  '\n'
[L:  3, P:  1]      |    unparsable:                                               !! Expected: "<Delimited: [<Ref: 'StatementSegment'>]>"
[L:  3, P:  1]      |        word:                                                 'from'
[L:  3, P:  5]      |        whitespace:                                           ' '
[L:  3, P:  6]      |        word:                                                 'tab'
[L:  3, P:  9]      |        semicolon:                                            ';'
[L:  3, P: 10]      |    newline:                                                  '\n'
[L:  4, P:  1]      |    [META] end_of_file:

==== parsing violations ====
L:   3 | P:   1 |  PRS | Line 3, Position 1: Found unparsable section: 'from tab;'
WARNING: Parsing errors found and dialect is set to 'ansi'. Have you configured your dialect correctly?
```

With the inline configuration:
```sh
> sqlfluff parse issues/66/sqlfluff_6657/test.sql
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    inline_comment:                                           '-- duckdb specific syntax'
[L:  1, P: 26]      |    newline:                                                  '\n'
[L:  2, P:  1]      |    inline_comment:                                           '-- sqlfluff:dialect:duckdb'
[L:  2, P: 27]      |    newline:                                                  '\n'
[L:  3, P:  1]      |    statement:
[L:  3, P:  1]      |        select_statement:
[L:  3, P:  1]      |            from_clause:
[L:  3, P:  1]      |                keyword:                                      'from'
[L:  3, P:  5]      |                whitespace:                                   ' '
[L:  3, P:  6]      |                from_expression:
[L:  3, P:  6]      |                    [META] indent:
[L:  3, P:  6]      |                    from_expression_element:
[L:  3, P:  6]      |                        table_expression:
[L:  3, P:  6]      |                            table_reference:
[L:  3, P:  6]      |                                naked_identifier:             'tab'
[L:  3, P:  9]      |                    [META] dedent:
[L:  3, P:  9]      |    statement_terminator:                                     ';'
[L:  3, P: 10]      |    newline:                                                  '\n'
[L:  4, P:  1]      |    [META] end_of_file:
```

If you have an example query of this not working, please feel free to share so I can investigate.
